src/Init/Data/List/Basic.lean

```diff
@@ -227,6 +227,12 @@ def findSome? (f : α → Option β) : List α → Option β
     | some b => some b
     | none   => findSome? f as
 
+def indexOf? [BEq α] (s : α) (i := 0): List α → Option Nat
```

Comment:
I think this parameter `(i := 0)` needs to be in an auxiliary, otherwise it will need to be passed explicitly when you use the function without dot notation.

Also there should be a space after the binder.

Reply:
I added indexOfAux?, but didn't understand the second comment "binding" ? what binder?

Reply:
And how do you hide the Aux methods from intellisense in VS code ?  I tried making it private but that didn't seem to help...

![image](https://user-images.githubusercontent.com/18707114/187048838-40b9aad0-3447-48fd-ba95-8cbe3717dd36.png)

Reply:
There's no need to hide the method, it's legal for people to call. If you use an inner function it will get a name like `indexOf?.go`, not sure how that shows up in the dropdown.

Also note that you are getting text suggestions there (as indicated by the "abc"); this is some vscode magic sauce that involves things you have written recently and words in the file so I wouldn't put too much stock in it. The better question is what the lean suggestions look like (and why you aren't getting them in that case).

The binder `(i := 0)` needs a space after it, i.e. `(i := 0) :`. It probably doesn't matter after the other change anyway.